code/script.js

```diff
@@ -0,0 +1,213 @@
+		if (Object.keys(authors).length > 1) {
+			document.getElementById(
+				`collaborators-${repo.name}`
+			).innerHTML += /*html*/ `<a href="${authors[author].html_url}" target="_blank"><img class="avatar-collaborator" src="${authors[author].avatar_url}"></a>`;
+		} else {
+			document.getElementById(`collaborators-${repo.name}`).innerHTML = /*html*/ 'Individual project';
+		}
+	}
+};
+
+const fetchPullRequestsArray = (repo, authors) => {
+	// fetch all pull requests from repo
+	const PULL_URL = `https://api.github.com/repos/${PARENT_OWNER}/${repo.name}/pulls?per_page=100`;
+	fetch(PULL_URL)
+		.then((res) => res.json())
+		.then((data) => {
+			// only pick pull requests connected to user
+			const myPullReq = data.find((pull) => authors.includes(pull.user.login));
+			if (myPullReq) {
+				pullReqData.done++;
+				reposData[repo.name].pullRequest = myPullReq;
+				updatePieChart(pieChart, pullReqData.done);
+			}
+			populatePullRequest(myPullReq, repo);
+		})
+		.catch((err) => alert('fetchPullRequestsArray error:', err));
+};
+
+const populatePullRequest = (myPullReq, repo) => {
+	if (myPullReq) {
+		document.getElementById(`pull-${repo.name}`).innerHTML = /*html*/ `<a href=${myPullReq.html_url} target="_blank">Pull request</a>`;
+	} else {
+		document.getElementById(`pull-${repo.name}`).innerHTML = /*html*/ 'No pull request done :(';
+	}
+};
+
+const fetchUser = () => {
+	fetch(USER_URL)
+		.then((res) => res.json())
+		.then((data) => {
+			userData.innerHTML += /*html*/ `<a href="${data.html_url}" target="_blank"><img class="avatar-user" src="${data.avatar_url}"></a><p class="user-name">${data.login}</p>`;
+		})
+		.catch((err) => alert('fetchCommits error: ', err));
+};
+
+const sortRepos = (array, param, init) => {
+	array.sort((a, b) => {
+		if (a[param] > b[param]) {
+			return -1;
+		} else if (a[param] < b[param]) {
+			return 1;
+		} else {
+			return 0;
+		}
+	});
+	if (param === 'name') array.reverse();
+	if (!init) regenerateProjectCards(reposArr);
+};
+
+const sortReposByCommit = (array, param) => {
+	array.sort((a, b) => {
+		if (reposData[a[param]].commits > reposData[b[param]].commits) {
+			return -1;
+		} else if (reposData[a[param]].commits < reposData[b[param]].commits) {
+			return 1;
+		} else {
+			return 0;
+		}
+	});
+	regenerateProjectCards(reposArr);
+};
+
+const filterRepos = (array, lang) => {
+	if (lang !== 'All') {
+		const filteredReposArr = array.filter((repo) => repo.language === lang);
+		regenerateProjectCards(filteredReposArr);
+	} else {
+		regenerateProjectCards(array);
+	}
+};
+
+const regenerateProjectCards = (repos) => {
+	projects.innerHTML = '';
+	repos.forEach((repo) => {
+		generateProjectCard(repo);
+		populateCommits(repo, reposData[repo.name].commits);
+		populateCollaborators(reposData[repo.name].authors, repo);
+		populatePullRequest(reposData[repo.name].pullRequest, repo);
+	});
+};
+
+sortSizeBtn.addEventListener('click', () => sortRepos(reposArr, 'size', false));
```

Comment:
feels like you implemented alot of what we learned when we made the guess who game. great to see that you reuse the knowledge and do so with ease